lib/sidekiq/metrics/queues.rb

```diff
@@ -0,0 +1,55 @@
+require 'sidekiq/api'
+
+module Sidekiq
+  module Metrics
+    class Queues
+      def initialize(
+        influxdb_client:,
+        series_name: 'sidekiq_queues',
+        retention_policy: nil,
+        tags: {},
+        clock: -> { Time.now.to_f }
+      )
+        @influxdb = influxdb_client
+        @series = series_name
+        @retention = retention_policy
+        @tags = tags
+        @clock = clock
+      end
+
+      def call
+        queues ||= ::Sidekiq::Stats::Queues.new.lengths
+
+        queues.each do |queue, size|
+          save(
+            tags: {queue: queue}.merge(@tags),
+            values: {size: size},
+            timestamp: in_correct_precision(@clock.call)
```

Comment:
Is it really necessary to set the timestamp explicitly here?

The middleware sends the timestamp because it wants to capture the exact moment when a job starts or finishes. If the middleware will not set the timestamps, the user will only find out time when metrics arrived to the InfluxDB server. It could be off by hundreds of milliseconds. It is important there.

But here, time is not very important, because metrics are gathered at HUGE (compared to jobs) intervals.